vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -0,0 +1,30 @@
+module SecureRooms
+
+  module AccessHandlers
+
+    class OrderHandler
+
+      def self.process(occupancy)
+        return unless occupancy.orderable?
+
+        order = Order.create!(
+          account: occupancy.account,
+          user: occupancy.user,
+          facility: occupancy.facility,
+          created_by_user: occupancy.user,
+        )
+
+        order.order_details.create!(
+          product: occupancy.secure_room,
+          created_by_user: occupancy.user,
+          quantity: 1,
+        )
+
+        order
```

Comment:
We usually create a module like `OrderDetailExtension` inside the engine, then in the engine's `to_prepare` block, add `OrderDetail.send :include, SecureRooms::OrderDetailExtension`. Take a look at the projects engine for a good example.